Simulation Code/Base/Base.ino

```diff
@@ -2,46 +2,63 @@
 #include "checkWaterLevel.h"
 #include "checkPumpStatus.h"
 
-// Define the pins for the water level sensors
-const int WaterLevelA = 4;
-const int WaterLevelB = 5;
-const int WaterLevelC = 6;
-const int WaterLevelD = 7;
-
-// Define the pins for the fans
-const int FanA = 9;
-const int FanB = 10;
-int FanPWM; // Variable to control the speed of the fans
-
-// Define the variables for the water pump
-unsigned long previousTime = 0; // Variable to store the last time the water level was checked
-unsigned long previousPumpTime = 0; // Variable to store the last time the pump status was checked
-const unsigned long stabilityTime = 1000; // Time in milliseconds to wait before changing the pump speed
-bool pumpStatus = false; // Variable to store the current status of the pump (true = on, false = off)
-const unsigned long pumpTimeOff = 1000; // Time in milliseconds for the pump to stay off
-const unsigned long pumpTimeOn = 5000; // Time in milliseconds for the pump to stay on
-const int WaterPump = 11; // Pin for the water pump
-int PumpPWM; // Variable to control the speed of the water pump
+// Define the structs
+
+struct WaterLevelSensors {
+  const int WaterPinA = 4;
+  const int WaterPinB = 5;
+  const int WaterPinC = 6;
+  const int WaterPinD = 7;
+  int waterLevelA;
+  int waterLevelB;
+  int waterLevelC;
+  int waterLevelD;
+};
+
+struct Fans {
+  const int FanA = 9;
+  const int FanB = 10;
+  int FanPWM;
+};
+
+struct WaterPumpFn {
+  unsigned long previousTime = 0;
+  unsigned long previousPumpTime = 0;
+  const unsigned long stabilityTime = 1000;
+  bool pumpStatus = false;
+  const unsigned long pumpTimeOff = 1000;
+  const unsigned long pumpTimeOn = 5000;
+  const int WaterPump = 11;
+  int PumpPWM;
+  unsigned long currentTime, elapsedTime, currentPumpTime, elapsedPumpTime;
+};
```

Comment:
Is it possible to have these structures defined in its own header file. One specifically for structures?

Reply:
not sure what you mean by this